.github/workflows/release.yml

```diff
@@ -132,27 +132,44 @@ jobs:
 
   windows:
     needs: setup
-    runs-on: windows-2022
+    strategy:
+      matrix:
+        include:
+          - runner: windows-latest
+            arch: x86_64
+            zip_name: nvim-win-x64.zip
+          - runner: windows-11-arm
+            arch: arm64
+            zip_name: nvim-win-arm64.zip
+    runs-on: ${{ matrix.runner }}
     steps:
       - uses: actions/checkout@v5
         with:
           # Perform a full checkout #13471
           fetch-depth: 0
       - run: .github/scripts/env.ps1
+        env:
+          BUILD_ARCH: ${{ matrix.arch }}
+      - name: Install Wix
+        run: |
+          Invoke-WebRequest -Uri "https://github.com/wixtoolset/wix3/releases/download/wix3141rtm/wix314-binaries.zip" -OutFile "wix314-binaries.zip"
+          Expand-Archive -Path "wix314-binaries.zip" -DestinationPath "C:/wix"
+          echo "C:\wix" >> $env:GITHUB_PATH
       - name: Build deps
         run: |
           cmake -S cmake.deps -B .deps -G Ninja -DCMAKE_BUILD_TYPE=${{ needs.setup.outputs.build_type }}
           cmake --build .deps
-      - name: build package
+      - name: Build package
         run: |
-          cmake -B build -G Ninja -DCMAKE_BUILD_TYPE=${{ needs.setup.outputs.build_type }}
+          cmake -B build -G Ninja -DCMAKE_BUILD_TYPE=${{ needs.setup.outputs.build_type }} -DCMAKE_SYSTEM_PROCESSOR=${{ matrix.arch }}
```

Comment:
do not pass `CMAKE_SYSTEM_PROCESSOR`, that is confusing. `CMAKE_SYSTEM_PROCESSOR` is a *cmake-provided* name. The cmake logic should check that, because that's also what users will be using when building locally. The logic should check for whatever value cmake provides on Windows x64/arm64. There is no reason to override it here, cmake already will tell us an appropriate value (and we can use `MATCHES` as already discussed).